File: backend/app/schemas/analytics.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field


class AnalyticsEventIn(BaseModel):
    event_type: str = Field(..., max_length=50)
    path: Optional[str] = Field(None, max_length=500)
    product_slug: Optional[str] = Field(None, max_length=200)
    session_id: Optional[str] = Field(None, max_length=80)
    utm_source: Optional[str] = Field(None, max_length=200)
    utm_medium: Optional[str] = Field(None, max_length=200)
    utm_campaign: Optional[str] = Field(None, max_length=200)

File: frontend/lib/api/analytics.ts
const API_BASE = process.env.NEXT_PUBLIC_API_BASE_URL || "https://api.officialskinksa.store";

export type StoreAnalyticsEventType =
  | "page_view"
  | "product_view"
  | "add_to_cart"
  | "begin_checkout";

export interface TrackStoreEventInput {
  event_type: StoreAnalyticsEventType;
  path?: string | null;
  product_slug?: string | null;
  session_id?: string | null;
  utm_source?: string | null;
  utm_medium?: string | null;
  utm_campaign?: string | null;
}

/** Fire-and-forget; never throws to the UI */
export function trackStoreEvent(payload: TrackStoreEventInput): void {
  if (typeof window === "undefined") return;
  void fetch(`${API_BASE}/v1/analytics/events`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  }).catch(() => {});
}
